### DISCUSSION TITLE
How to debug "A cycle occurred while laying the GUI"

### DISCUSSION THREAD

**Discussion by DanTravison** [dotnet/maui#15709] [Q&A]
I hit this periodically and, frankly, it is undiagnosable.  The callstack provides no useful information and the exception itself is generic.

When I encounter it, I end up trying alternative layouts to avoid it but I would prefer to be able to see some useful ideas to either avoid it or point to improper coding patterns that may be causing it.

Has anyone found a method for diagnosing this or any coding patterns that may be contributing to it?

Thanks,
Dan.

**Comment by daltzctr** [comment]
I've found testing on Android to show the real exception in these cases, or to not fault at all.

Another strategy I've done is

- nuke layout until it works
- add layout back, starting from outer to inner
- see crash, find workaround

It sucks, but welcome to MAUI.

**Comment by hartez** [comment]
If you're seeing this error, please open an issue and post the layout which is causing it. 

Unfortunately, this error comes from the layout cycle detection in the WinUI layer, so we don't have any control over the amount of information it gives you. Debugging it requires stepping through the layout code in MAUI or subclassing all of your layouts and overriding `MeasureOverride` and `ArrangeOverride` so you can see where the repeated layout calls are happening.

**Reply by daltzctr** [reply]
Both of those situations have no documentation AFAIK.
